packages/cursorless-vscode-core/src/compositionRoot/createVscodeIde.ts

```diff
@@ -0,0 +1,10 @@
+import * as vscode from "vscode";
+import { VscodeHats, VscodeIDE } from "..";
+
+export async function createVscodeIde(context: vscode.ExtensionContext) {
+  const vscodeIDE = new VscodeIDE(context);
+
+  const hats = new VscodeHats(vscodeIDE, context);
+  await hats.init();
+  return { vscodeIDE, hats };
```

Comment:
also experimenting with letting the VSCode ide code have its own factory / composition root.  Then the only tight coupling within the vscode ide is here, rather than in the old approach where we had one big class that instantiates everything